### ISSUE TITLE
[Bug]: New files are not generated by shopify theme push from CI/CD

### ISSUE THREAD

**Issue by Hyuga-Niigaki** [Shopify/cli#3636]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Execute bundle task, new bundle files are generated (custom.css , custom.js) and these are reflected in theme.

I was wondering if there was a problem with the files generated by the bundle.
However,
the anima.js is not a file generated by the bundle, but a newly created file in the theme folder, which should also be reflected in the theme.
![スクリーンショット 2024-04-02 14 19 43](https://github.com/Shopify/cli/assets/110234229/afdd2c41-ff8f-4d1f-ac4d-bc3900a27fdc)

<img width="1440" alt="スクリーンショット 2024-04-02 14 09 30" src="https://github.com/Shopify/cli/assets/110234229/52bf52a2-af2c-4213-83ee-d29c91c2291e">

### Actual behavior

All newly generated files are not reflected in theme.

### Verbose output

```shell
Run shopify theme push --allow-live --theme=*** --json --verbose
  shopify theme push --allow-live --theme=*** --json --verbose
  shell: /usr/bin/bash -e {0}
  env:
    SHOPIFY_FLAG_STORE: ***
    SHOPIFY_CLI_THEME_TOKEN: ***
    SHOPIFY_CLI_TTY: 0
2024-04-01T15:16:39.473Z: Running command theme push
2024-04-01T15:16:39.479Z shopify:theme:push init version: @oclif/core@3.19.6 argv: [ '--allow-live', '--theme=***', '--json', '--verbose' ]
2024-04-01T15:16:39.493Z: Ensuring that the user is authenticated with the Theme API with the following scopes:
[]

2024-04-01T15:16:39.499Z: Getting development theme...
2024-04-01T15:16:40.139Z: Request to https://theme-kit-access.shopifyapps.com/cli/admin/api/unstable/themes.json?fields=id%2Cname%2Crole%2Cprocessing completed in 632 ms
With response headers:
 - cache-control: max-age=0, private, must-revalidate
 - content-type: application/json; charset=utf-8
 - etag: W/"ad4e6fba3eb83066d467f0bf9ccc6669"
 - x-request-id: c15ec0f7-84f4-4b06-985e-7de7e285a30c-1711984599
    
2024-04-01T15:16:40.827Z: Request to https://theme-kit-access.shopifyapps.com/cli/admin/api/unstable/themes/***/assets.json?fields=key%2Cchecksum completed in 680 ms
With response headers:
 - cache-control: max-age=0, private, must-revalidate
 - content-type: application/json; charset=utf-8
 - etag: W/"6479bc207fbd3ec26edaa4f67e0b178a"
 - x-request-id: 2a4cfb28-95c5-4211-abab-fa26a7a440a8-1711984600
    
2024-04-01T15:16:40.866Z: Reading the content of file at assets/anima.js...
2024-04-01T15:16:40.866Z: Reading the content of file at assets/base.css...
2024-04-01T15:16:40.866Z: Reading the content of file at assets/cart-drawer.js...
2024-04-01T15:16:40.867Z: Reading the content of file at assets/cart-notification.js...
2024-04-01T15:16:40.867Z: Reading the content of file at assets/cart.js...
2024-04-01T15:16:40.867Z: Reading the content of file at assets/collage.css...
2024-04-01T15:16:40.868Z: Reading the content of file at assets/collapsible-content.css...
2024-04-01T15:16:40.868Z: Reading the content of file at assets/component-accordion.css...
2024-04-01T15:16:40.868Z: Reading the content of file at assets/component-article-card.css...
2024-04-01T15:16:40.868Z: Reading the content of file at assets/component-card.css...
2024-04-01T15:16:40.869Z: Reading the content of file at assets/component-cart-drawer.css...
2024-04-01T15:16:40.869Z: Reading the content of file at assets/component-cart-items.css...
2024-04-01T15:16:40.869Z: Reading the content of file at assets/component-cart-notification.css...
2024-04-01T15:16:40.869Z: Reading the content of file at assets/component-cart.css...
2024-04-01T15:16:40.869Z: Reading the content of file at assets/component-collection-hero.css...
2024-04-01T15:16:40.870Z: Reading the content of file at assets/component-complementary-products.css...
2024-04-01T15:16:40.870Z: Reading the content of file at assets/component-deferred-media.css...
2024-04-01T15:16:40.872Z: Reading the content of file at assets/component-discounts.css...
2024-04-01T15:16:40.872Z: Reading the content of file at assets/component-facets.css...
2024-04-01T15:16:40.872Z: Reading the content of file at assets/component-image-with-text.css...
2024-04-01T15:16:40.873Z: Reading the content of file at assets/component-list-menu.css...
2024-04-01T15:16:40.873Z: Reading the content of file at assets/component-list-payment.css...
2024-04-01T15:16:47.881Z: Obtaining the dependency manager in directory /home/runner/work/custmizeEC/custmizeEC...
2024-04-01T15:16:47.887Z: 
Running system process:
  · Command: ruby -v
  · Working directory: /home/runner/work/custmizeEC/custmizeEC

2024-04-01T15:16:48.170Z: Request to https://monorail-edge.shopifysvc.com/v1/produce completed in 259 ms
With response headers:
 - x-request-id: fb2cdb40-0d5d-476e-bc9d-2540c987cce3
    
2024-04-01T15:16:48.171Z: Analytics event sent: {
  "command": "theme push",
  "time_start": 1711984599474,
  "time_end": 1711984607878,
  "total_time": 8404,
  "success": true,
  "cli_version": "3.58.0",
  "ruby_version": "3.1.4",
  "node_version": "18.19.1",
  "is_employee": false,
  "uname": "linux amd64",
  "env_ci": true,
  "env_ci_platform": "github",
  "env_plugin_installed_any_custom": false,
  "env_plugin_installed_shopify": "[\"@shopify/cli\",\"@shopify/plugin-did-you-mean\",\"@shopify/theme\"]",
  "env_shell": "bash",
  "env_device_id": "808dfaefc23074c20c200e756c6c4c159ea9d44e",
  "env_cloud": "localhost",
  "env_package_manager": "npm",
  "cmd_all_timing_network_ms": 0,
  "cmd_all_timing_prompts_ms": 0,
  "cmd_all_launcher": "unknown",
  "cmd_all_topic": "theme",
  "cmd_all_plugin": "@shopify/theme",
  "cmd_all_force": false,
  "cmd_all_timing_active_ms": 8404,
  "cmd_all_exit": "ok",
  "args": "--allow-live --theme=*** --json --verbose",
  "env_plugin_installed_all": "[\"@shopify/cli\",\"@shopify/plugin-did-you-mean\",\"@shopify/theme\"]",
  "metadata": "{\"extraPublic\":{},\"extraSensitive\":{}}"
}
2024-04-01T15:16:48.171Z: Completed command theme push
```

### Reproduction steps

1. Create the following github actions

```
name: Build and Deploy Theme

on:
  push:
    branches:
      - main

jobs:
  build-and-push:
    if: github.actor != 'shopify[bot]'
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
        with:
          ref: main
      - uses: actions/setup-node@v3
        with:
          node-version: 18
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: 3.1
          bundler: 'latest'
      - name: Install Shopify CLI
        run: npm install -g @shopify/cli @shopify/theme
      - name: Install dependencies
        run: npm install
      - name: Build
        run: |
         npm run build
         echo "Searching for custom.css..."
         find . -name "custom.css"
      - name: Deploy
        env:
          SHOPIFY_FLAG_STORE: '${{ secrets.SHOP_STORE_OS2 }}'
          SHOPIFY_CLI_THEME_TOKEN: ${{ secrets.SHOPIFY_CLI_THEME_TOKEN }}
          SHOPIFY_CLI_TTY: 0
        run: shopify theme push --allow-live --theme=${{ secrets.SHOPIFY_THEME_ID }} --json --verbose
```

2. Create a new file and push it to the main branch
3. Check to see if the newly generated files from the remote theme are reflected

### Operating System

ubuntu-latest

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.58

### Node version (run `node -v` if you're not sure)

18.19.1

### What language and version are you using in your application?

"ruby_version": "3.1.4"

**Comment by xmadscientist** [comment]
I am experiencing a similar issue with just the standard `shopify theme push ...` command in the CLI without the CI/CD stuff. 

Has anyone found a resolution to this? This only happened after I upgraded the Shopify CLI to the new version.

**Comment by jamesmengo** [comment]
This was fixed in release `3.58.2`
You can either upgrade to `3.58.2` or use the `--stable` flag :) 

- https://github.com/Shopify/cli/pull/3645

**Comment by min23a** [comment]
can anyone help me with this-
remote: Enumerating objects: 23398, done.                                                                         │
remote: Counting    │cts:   1% (2/143)
remote: Counting objects:   4%       │        
remote: Counting objects:   6% (9/143)             │
remote: Counting │bjects:   8% (12/143)        
remote: Counting objects:  11%     │43)        
remote: Counting objects:  13% (19/143)          │
remote: Counting │bjects:  15% (22/143)        
remote: Counting objects:  18%     │43)        
remote: Counting objects:  20% (29/143)          │
remote: Counting │bjects:  22% (32/143)        
remote: Counting objects:  25%     │43)
remote: Counting objects:  27% (39/143)          │
remote: Counting │bjects:  29% (42/143)
remote: Counting objects:  32%     │43)
remote: Counting objects:  34% (49/143)          │
remote: Counting │bjects:  36% (52/143)
remote: Counting objects:  39%     │43)
remote: Counting objects:  41% (59/143)          │
remote: Counting │bjects:  43% (62/143)
remote: Counting objects:  46%     │43)
remote: Counting objects:  48% (69/143)          │
remote: Counting │bjects:  50% (72/143)
remote: Counting objects:  53%     │43)
remote: Counting objects:  55% (79/143)          │
remote: Counting │bjects:  57% (82/143)
remote: Counting objects:  60%     │43)
remote: Counting objects:  62% (89/143)          │
remote: Counting │bjects:  64% (92/143)
remote: Counting objects:  67%     │43)
remote: Counting objects:  69% (99/143)          │
remote: Counting objects:  71% (102/143)
remote: Counting objects:  74%  │05/143)
remote: Counting objects:  76% (109/143)       │
remote:       │g objects:  78% (112/143)
remote: Counting        │  80% (115/143)
remote: Counting objects:  83%   │8/143)
remote: Counting objects:  85% (122/143)       │
remote:       │g objects:  87% (125/143)
remote: Counting        │  89% (128/143)
remote: Counting objects:  92%   │1/143)
remote: Counting objects:  94% (135/143)       │
remote:       │g objects:  96% (138/143)
remote: Counting        │  98% (141/143)
remote: Counting objects: 100%   │3/143)
│  (143/143), done.                                                                                                  │
remote:         │ng objects:   3% (2/65)
remote: Compressing     │ts:   6% (4/65)
remote: Compressing objects:  10%   │65)
remote: Compressing objects:  13% (9/65)          │
remote:      │ssing objects:  16% (11/65)
remote: Compressing   │ects:  20% (13/65)
remote: Compressing objects:  24% │15/65)
remote: Compressing objects:  27% (18/65)       │
remote:     │essing objects:  30% (20/65)
remote: Compressing   │ects:  33% (22/65)
remote: Compressing objects:  38% │24/65)
remote: Compressing objects:  41% (27/65)       │
remote:     │essing objects:  44% (29/65)
remote: Compressing   │ects:  47% (31/65)
remote: Compressing objects:  52% │33/65)
remote: Compressing objects:  55% (36/65)       │
remote:     │essing objects:  58% (38/65)
remote: Compressing   │ects:  61% (40/65)
remote: Compressing objects:  66% │42/65)
remote: Compressing objects:  69% (45/65)       │
remote:     │essing objects:  72% (47/65)
remote: Compressing   │ects:  75% (49/65)
remote: Compressing objects:  80% │51/65)
remote: Compressing objects:  83% (54/65)       │
remote:     │essing objects:  86% (56/65)
remote: Compressing   │ects:  89% (58/65)
remote: Compressing objects:  93% │60/65)
remote: Compressing objects:  96% (63/65)       │
remote:     │essing objects: 100% (65/65)
│  Compressing objects: 100% (65/65), done.                                                                          │
Receiving objects: │ 1% (234/23398), 84.00 KiB | 92.00 KiB/s
Receiving │bjects:   1% (346/23398), 180.00 KiB | 74.00 KiB/s
Receiving objects:   2% (468/23398), 244.00 KiB | 68.00       │
Receiving objects:   2% (616/23398), 300.00   │ | 63.00 KiB/s
Receiving objects:   3%           │, 348.00 KiB | 58.00 KiB/s
Receiving     │ts:   3% (889/23398), 428.00 KiB | 55.00 KiB/s
Receiving objects:   4% (1086/23398), 524.00 KiB | 59.00      │
Receiving objects:   5% (1192/23398), 556.00 │iB | 56.00 KiB/s
Receiving objects:   6%          │8), 604.00 KiB | 57.00 KiB/s
Receiving   │ects:   6% (1537/23398), 684.00 KiB | 63.00 KiB/s
Receiving objects:   7% (1711/23398), 756.00 KiB | 64.00     │
Receiving objects:   8% (1872/23398), 796.00 │iB | 63.00 KiB/s
Receiving objects:   8%          │8), 836.00 KiB | 56.00 KiB/s
Receiving   │ects:   9% (2106/23398), 892.00 KiB | 53.00 KiB/s
Receiving objects:   9% (2246/23398), 964.00 KiB | 44.00     │
Receiving objects:  10% (2341/23398), 996.00 │iB | 44.00 KiB/s
Receiving objects:  11%          │8), 1.03 MiB | 39.00 KiB/s
Receiving       │:  11% (2649/23398), 1.11 MiB | 45.00 KiB/s
Receiving objects:  11% (2807/23398), 1.18 MiB | 49.00         │
Receiving objects:  12% (2870/23398), 1.24 MiB │ 49.00 KiB/s
Receiving objects:  12% (3002/23398),  │25 MiB | 48.00 KiB/s
Receiving objects:  13%       │3398), 1.33 MiB | 42.00 KiB/s
Receiving       │:  13% (3063/23398), 1.43 MiB | 43.00 KiB/s
Receiving objects:  13% (3064/23398), 1.50 MiB | 46.00         │
Receiving objects:  13% (3225/23398), 1.68 MiB │ 55.00 KiB/s
Receiving objects:  14% (3276/23398),  │82 MiB | 85.00 KiB/s
Receiving objects:  14%      │23398), 1.98 MiB | 110.00 KiB/s
Receiving     │ts:  14% (3362/23398), 2.22 MiB | 126.00 KiB/s
Receiving objects:  14% (3457/23398), 2.42 MiB | 124.00       │
Receiving objects:  14% (3492/23398), 2.59 MiB│| 111.00 KiB/s
Receiving objects:  15%             │ 2.64 MiB | 96.00 KiB/s
Receiving       │:  15% (3557/23398), 2.79 MiB | 93.00 KiB/s
Receiving objects:  15% (3626/23398), 2.98 MiB | 98.00         │
Receiving objects:  16% (3744/23398), 3.17 MiB│| 106.00 KiB/s
Receiving objects:  16%             │ 3.23 MiB | 114.00 KiB/s
Receiving     │ts:  16% (3848/23398), 3.43 MiB | 106.00 KiB/s
Receiving objects:  16% (3901/23398), 3.57 MiB | 94.00 KiB/s  │
Receiving objects:  16% (3939/23398), 3.71 MiB |    │0 KiB/s
Receiving objects:  17% (3999/23398),    │ MiB | 69.00 KiB/s
Receiving objects:  17%       │3398), 3.83 MiB | 68.00 KiB/s
Receiving       │:  17% (4067/23398), 4.01 MiB | 71.00 KiB/s
Receiving objects:  17% (4119/23398), 4.14 MiB | 78.00         │
Receiving objects:  17% (4204/23398), 4.29 MiB │ 82.00 KiB/s
Receiving objects:  18% (4236/23398),  │29 MiB | 75.00 KiB/s
Receiving objects:  18%       │3398), 4.38 MiB | 64.00 KiB/s
Receiving       │:  18% (4259/23398), 4.47 MiB | 46.00 KiB/s
Receiving objects:  18% (4288/23398), 4.57 MiB | 46.00         │
Receiving objects:  18% (4361/23398), 4.74 MiB │ 54.00 KiB/s
Receiving objects:  18% (4390/23398),  │82 MiB | 70.00 KiB/s
Receiving objects:  18%       │3398), 4.98 MiB | 82.00 KiB/s
Receiving       │:  19% (4446/23398), 5.11 MiB | 83.00 KiB/s
Receiving objects:  19% (4491/23398), 5.19 MiB | 76.00         │
Receiving objects:  19% (4524/23398), 5.33 MiB │ 71.00 KiB/s
Receiving objects:  19% (4599/23398),  │39 MiB | 61.00 KiB/s
Receiving objects:  20%       │3398), 5.47 MiB | 55.00 KiB/s
Receiving       │:  20% (4739/23398), 5.54 MiB | 54.00 KiB/s
Receiving objects:  20% (4798/23398), 5.64 MiB | 56.00         │
Receiving objects:  20% (4819/23398), 5.79 MiB │ 60.00 KiB/s
Receiving objects:  20% (4871/23398),  │82 MiB | 62.00 KiB/s
Receiving objects:  21%       │3398), 6.00 MiB | 68.00 KiB/s
Receiving       │:  21% (4945/23398), 6.08 MiB | 71.00 KiB/s
Receiving objects:  21% (4982/23398), 6.24 MiB | 77.00         │
Receiving objects:  21% (5039/23398), 6.40 MiB │ 78.00 KiB/s
Receiving objects:  21% (5074/23398),  │51 MiB | 87.00 KiB/s
Receiving objects:  21%      │23398), 6.76 MiB | 103.00 KiB/s
Receiving     │ts:  21% (5137/23398), 6.93 MiB | 112.00 KiB/s
Receiving objects:  22% (5148/23398), 7.12 MiB | 119.00       │
Receiving objects:  22% (5212/23398), 7.30 MiB│| 119.00 KiB/s
Receiving objects:  22%             │ 7.41 MiB | 123.00 KiB/s
Receiving     │ts:  22% (5271/23398), 7.63 MiB | 107.00 KiB/s
Receiving objects:  22% (5297/23398), 7.76 MiB | 96.00 KiB/s  │
Receiving objects:  22% (5318/23398), 7.90 MiB |    │0 KiB/s
Receiving objects:  22% (5338/23398),    │ MiB | 65.00 KiB/s
Receiving objects:  22%       │3398), 8.04 MiB | 55.00 KiB/s
Receiving       │:  23% (5382/23398), 8.15 MiB | 54.00 KiB/s
Receiving objects:  23% (5394/23398), 8.23 MiB | 57.00         │
Receiving objects:  23% (5446/23398), 8.36 MiB │ 59.00 KiB/s
Receiving objects:  23% (5454/23398),  │37 MiB | 57.00 KiB/s
Receiving objects:  23%       │3398), 8.50 MiB | 50.00 KiB/s
Receiving       │:  23% (5586/23398), 8.82 MiB | 98.00 KiB/s
Receiving objects:  24% (5643/23398), 9.06 MiB | 138.00        │
Receiving objects:  24% (5727/23398), 9.46 MiB│| 162.00 KiB/s
Receiving objects:  24%             │ 9.68 MiB | 191.00 KiB/s
Receiving     │ts:  24% (5810/23398), 9.89 MiB | 164.00 KiB/s
Receiving objects:  24% (5842/23398), 10.08 MiB | 124.00     │
Receiving objects:  25% (5871/23398), 10.16  │B | 124.00 KiB/s
Receiving objects:  25%          │8), 10.22 MiB | 90.00 KiB/s
Receiving     │ts:  25% (5913/23398), 10.43 MiB | 87.00 KiB/s
Receiving objects:  25% (5939/23398), 10.57 MiB | 89.00       │
Receiving objects:  25% (5989/23398), 10.80   │ | 92.00 KiB/s
Receiving objects:  25%           │), 10.93 MiB | 96.00 KiB/s
Receiving   │ects:  25% (6071/23398), 11.14 MiB | 109.00 KiB/s
Receiving objects:  26% (6084/23398), 11.24 MiB | 108.00     │
Receiving objects:  26% (6155/23398), 11.41  │B | 93.00 KiB/s
Receiving objects:  26%          │8), 11.47 MiB | 103.00 KiB/s
Receiving   │ects:  26% (6261/23398), 11.76 MiB | 110.00 KiB/s
Receiving objects:  27% (6438/23398), 11.96 MiB | 142.00     │
Receiving objects:  28% (6552/23398), 12.18  │B | 158.00 KiB/s
Receiving objects:  28%         │98), 12.39 MiB | 181.00 KiB/s
Receiving   │ects:  29% (6786/23398), 12.74 MiB | 212.00 KiB/s
Receiving objects:  30% (7020/23398), 12.94 MiB | 210.00     │
Receiving objects:  30% (7215/23398), 13.22  │B | 223.00 KiB/s
Receiving objects:  31%         │98), 13.34 MiB | 226.00 KiB/s
Receiving   │ects:  32% (7488/23398), 13.69 MiB | 240.00 KiB/s
Receiving objects:  32% (7544/23398), 13.93 MiB | 224.00     │
Receiving objects:  33% (7722/23398), 14.20  │B | 222.00 KiB/s
error: RPC failed; curl 18      │98), 14.37 MiB | 233.00 KiB/s
│  transfer closed with outstanding read data remaining                                                              │
│  error: 7436 bytes of body are still expected                                                                      │
│  fetch-pack: unexpected disconnect while reading sideband packet                                                   │
│  fatal: early EOF                                                                                                  │
│  fatal: fetch-pack: invalid index-pack output
